epuck_ros2_camera/src/camera.cpp

```diff
@@ -0,0 +1,175 @@
+#include <chrono>
+#include <memory>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/compressed_image.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <opencv2/opencv.hpp>
+
+extern "C"
+{
+#include "epuck_ros2_camera/pipuck_image.h"
+#include "epuck_ros2_camera/pipuck_v4l2.h"
+#include "epuck_ros2_camera/pipuck_jpeg.h"
+#include "epuck_ros2_camera/pipuck_ov7670.h"
+}
+
+using namespace std::chrono_literals;
+
+class CameraPublisher : public rclcpp::Node
+{
+public:
+  CameraPublisher() : Node("camera_publisher"), v4l2_initialized(false), jpeg_initialized(false)
+  {
+    auto quality = declare_parameter<int>("quality", 8);
+    auto interval = declare_parameter<int>("interval", 80);
+
+    pipuck_image_init(&captured_image);
+    pipuck_image_init(&compressed_image);
+
+    compressed_image.quality = quality;
+    compressed_image.data = image_buffer;
+    captured_image.encoding = PIPUCK_IMAGE_ENCODING_YUYV;
+
+    pipuck_ov7670_init();
+
+    callback_handler = this->add_on_set_parameters_callback(std::bind(&CameraPublisher::param_change_callback, this, std::placeholders::_1));
+
+    publisher_compressed = this->create_publisher<sensor_msgs::msg::CompressedImage>(
+        "image_raw/compressed", 0);
+    publisher_raw = this->create_publisher<sensor_msgs::msg::Image>(
+        "image_raw", 0);
+    timer = this->create_wall_timer(
+        std::chrono::milliseconds(interval), std::bind(&CameraPublisher::timer_callback, this));
+  }
+
+  ~CameraPublisher()
+  {
+    if (v4l2_initialized == true) {
+      pipuck_v4l2_deinit();
+    }
+    if (jpeg_initialized == true) {
+      pipuck_jpeg_deinit();
+    }
```

Comment:
We usually avoid adding brackets for one line condition: https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style#cs360-single-if-while-or-for-statement-must-be-written-without-brackets-the-statement-part-should-be-put-on-a-separate-line-in-case-of-nested-block-keep-brackets

Reply:
> It is globally very good and clean, well done for a first PR!
> 
> There is just a few polishing issues:
> 
> Several files are missing the license headers.
> 
> There are a few formatting issues, it would be nice to use (maybe copy it in this repo) the general Webots clang-format file (it is ok for copied files if the format is not matching, but would be nice to respect it for files you wrote):
> https://github.com/cyberbotics/webots/blob/master/dependencies/.clang-format-6

Oh, this is fantastic, I didn't know you have `.clang-format`! I was configuring my own :/